### PR TITLE
Orion Trail Content Update

### LINKED ISSUE #10982: Arcade machine exploit
When you play the game Orion Trail, if you click continue more than once on continue it skips complete scenes and its actions .For example you click continue 3 times very fast , it skips some important acts like raiders stole all your food or gas , skips breakdowns and you can even skip the black hole part.At the ending screen if you click 3 times on the last continue you get 3x the prize.This can be used by traitors that emag the machine to farm tank transfer valves.Average game takes 10 seconds and you cant lose.Now instead of 6 maxcap bombs a traitor can have 30 in a matter of minutes.

### PR DESCRIPTION
**Pull request by RemieRichards**
## New Content!
- Spaceports!
- Buy/Sell crew members for Food and Fuel
- Trade Food for Fuel, and Fuel for Food
- Buy Spare Electronics, Engine Parts and Hull Plates
- Try and Raid the spaceport, you get free stuff, but is it worth it?
- Changeling themed Events!
- Changelings can stow away in your crew, they eat double food rations, pay attention to your food to spot them
- Changelings have a 75%/55% (it varies based on prior events) chance to turn the next event into a Changeling Ambush, removing them from your crew permanently, with an extra chance to kill some other crew, if you manage to kill the lings you can grind them up for food... Gross!
- You can now shoot any crew member in the head for NO REASON! (Well ok, Maybe you don't have enough food to take 4 people to Orion, or maybe you think they're a changeling!)
## Code
- Cleaned up Magicstrings, they're now #defines 
- Added some procs that handle fully adding/fully removing crewmembers
- Made the UI for Orion Trail bigger, because Spaceports have a lot of info/actions
- Adds /orion subtype of ranged syndicate mob, they're used as punishment for failing to raid a spaceport while emagged (it's not a guaranteed chance to happen, but enjoy 4 Syndie SMGs in your face)
- Fixes #10982

### CODE REVIEW

**Comment by Cheridan** [comment]
DEEPEST
LORE

**Comment by PKPenguin321** [comment]
:+1: :+1: :+1: :+1: 

not enough thumbs in the world for this one

**Comment by Incoming5643** [comment]
It's usually great to lose a crewmember of two on the trip because the food stores has very little wiggle room for any unexpected loses during the trip if everyone's eating all the time

**Comment by CosmicScientist** [comment]
:+1:

File: code/game/machinery/computer/arcade.dm [hunk 1]
```diff
@@ -584,6 +610,108 @@
 	else if(href_list["eventclose"]) //end an event
 		event = null
 
+	else if(href_list["killcrew"]) //shoot a crewmember
+		var/sheriff = remove_crewmember() //I shot the sheriff
```
Comment (Iamgoofball) [inline code/game/machinery/computer/arcade.dm]: what about the deputy

File: code/game/machinery/computer/arcade.dm [hunk 2]
```diff
@@ -584,6 +610,108 @@
...
+	else if(href_list["buycrew"]) //buy a crewmember
+		var/bought = add_crewmember()
+		last_spaceport_action = "You hired [bought] as a new crewmember."
+		fuel -= 10
+		food -= 10
+		event()
+
+	else if(href_list["sellcrew"]) //sell a crewmember
+		var/sold = remove_crewmember()
+		last_spaceport_action = "You sold your crewmember, [sold]!"
```
Comment (Iamgoofball) [inline code/game/machinery/computer/arcade.dm]: >slavery
>no reprecussions

File: code/game/machinery/computer/arcade.dm [hunk 3]
```diff
@@ -584,6 +610,108 @@
...
+			FO = rand(-5,-15)
+			last_spaceport_action = "You failed to raid the spaceport! you lost [FU*-1] Fuel and [FO*-1] Food in your scramble to escape! ([FU]FU,[FO]FO)"
+
+			//your chance of lose a crewmember is 1/2 your chance of success
+			//this makes higher % failures hurt more, don't get cocky space cowboy!
+			if(prob(success*5))
+				var/lost_crew = remove_crewmember()
+				last_spaceport_action = "You failed to raid the spaceport! you lost [FU*-1] Fuel and [FO*-1] Food, AND [lost_crew] in your scramble to escape! ([FU]FI,[FO]FO,-Crew)"
+				if(emagged)
+					say("WEEWOO WEEWOO, Spaceport Security en route!")
```
Comment (Iamgoofball) [inline code/game/machinery/computer/arcade.dm]: needs to play the WEE WOO WEE WOO sound

File: code/game/machinery/computer/arcade.dm (hunk 1 above)
Comment (RemieRichards) [inline code/game/machinery/computer/arcade.dm]: I did not shoot the deputy!

File: code/game/machinery/computer/arcade.dm (hunk 2 above)
Comment (RemieRichards) [inline code/game/machinery/computer/arcade.dm]: Well, you lose strength for the combat events, and come closer to death.
Same repercussions as just shooting the guy in the head, except you get paid.

also I was so engrained in the dangerously shitty world of Orion Trail that slavery was just natural :P

File: code/game/machinery/computer/arcade.dm [hunk 4]
```diff
@@ -591,81 +716,271 @@
...
+						eventdat += "<br>You valiantly fight off the Changeling[ling2 ? "s":""]!"
+						eventdat += "<br>You cut the Changeling[ling2 ? "s":""] up into meat... Eww"
+						if(ling2)
+							food += 30
+							lings_aboard = max(0,lings_aboard-2)
+						else
+							food += 15
+							lings_aboard = max(0,--lings_aboard)
+				else
+					eventdat += "<br>The Changeling[ling2 ? "s":""] run away, What wimps!"
```
Comment (Cheridan) [inline code/game/machinery/computer/arcade.dm]: So if there's 1 changeling, it says "The Changeling run away, What wimps!"

Even if you say this is a reference to changelings referring to themselves in the plural, it's still not very sound grammatically :P

Comment (RemieRichards) [inline code/game/machinery/computer/arcade.dm]: dang it, no, it's me missing the trinary statement on the word run as well, like I did further up.

**Comment by RemieRichards** [comment]
@Cheridan engrish collected.
:P

File: code/game/machinery/computer/arcade.dm [hunk 5]
```diff
@@ -591,81 +716,271 @@
...
+				eventdat += "Oh no, some of your crew are Changelings!"
+				if(ling2)
+					eventdat += "<br>[ling1] and [ling2]'s arms twist and contort into grotesque blades!"
+				else
+					eventdat += "<br>[ling1]'s arm twists and contorts into a grotesque blade!"
+
+				var/chance2attack = alive*20
+				if(chance2attack)
+					var/chancetokill = 30*lings_aboard-(5*alive) //eg: 30*2-(10) = 50%, 2 lings, 2 crew is 50% chance
+					if(chancetokill)
```
Comment (phil235) [inline code/game/machinery/computer/arcade.dm]: don't you mean prob(chancetokill) ?

File: code/game/machinery/computer/arcade.dm [hunk 6]
```diff
@@ -591,81 +716,271 @@
...
+					ling2 = remove_crewmember()
+
+				eventdat += "Oh no, some of your crew are Changelings!"
+				if(ling2)
+					eventdat += "<br>[ling1] and [ling2]'s arms twist and contort into grotesque blades!"
+				else
+					eventdat += "<br>[ling1]'s arm twists and contorts into a grotesque blade!"
+
+				var/chance2attack = alive*20
+				if(chance2attack)
```
Comment (phil235) [inline code/game/machinery/computer/arcade.dm]: don't you mean prob(chancetoattack) ?

**Comment by phil235** [comment]
Also, can you fix #10982 in this PR? Just add a busy var to the machine and in the topic proc check for busy, then do busy = 1 and busy = 0 at the very end of the proc. It fixes the issue.

File: code/game/machinery/computer/arcade.dm (hunk 5 above)
Comment (RemieRichards) [inline code/game/machinery/computer/arcade.dm]: Son of a biscuit.
Gosh, I know it was late when I wrote this but to mess up a prob? yiiiikes.

**Comment by RemieRichards** [comment]
@phil235 fixed the probs and added a busy check.
I allowed users to still close the game while it's busy, as that doesn't actually affect any of the game's contents so they won't be skipping anything.

**Comment by Jalleo** [comment]
have you got the fix for #10982 in this?

**Comment by phil235** [comment]
merge conflicts

File: code/game/machinery/computer/arcade.dm [hunk 7]
```diff
@@ -431,34 +453,45 @@
...
 					event = pickweight(events)
+					if(lings_aboard)
+						if(event == ORION_TRAIL_LING || prob(55))
+							event = ORION_TRAIL_LING_ATTACK
 					event()
 		else
-			food -= alive*2
+			food -= (alive+lings_aboard)*2
 			fuel -= 5
 			turns += 1
```
Comment (phil235) [inline code/game/machinery/computer/arcade.dm]: You can avoid having those three lines copypasted three times by changing `else if(turn ==2)` into
```
else
	food -= (alive+lings_aboard)*2
 	fuel -= 5
 	turns += 1
	if(turns == 2)
		if(prob(30))
		else
	else
		if(prob(75))

**Comment by RemieRichards** [comment]
For the love of fuck @tgstation/commit-access somebody merge this.

**Pull request merged**